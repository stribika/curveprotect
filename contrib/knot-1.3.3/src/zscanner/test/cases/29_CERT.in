$ORIGIN	.
$TTL	1

; OK
@	CERT	0	0	0	AA==		; The simplest variant
@	CERT	65535	65535	255	AA==		; Maximal numbers
@	CERT	PKIX	0	0	AA==		; Certificate type mnemo
@	CERT	SPKI	0	0	AA==		; Certificate type mnemo
@	CERT	PGP	0	0	AA==		; Certificate type mnemo
@	CERT	IPKIX	0	0	AA==		; Certificate type mnemo
@	CERT	ISPKI	0	0	AA==		; Certificate type mnemo
@	CERT	IPGP	0	0	AA==		; Certificate type mnemo
@	CERT	ACPKIX	0	0	AA==		; Certificate type mnemo
@	CERT	IACPKIX	0	0	AA==		; Certificate type mnemo
@	CERT	URI	0	0	AA==		; Certificate type mnemo
@	CERT	OID	0	0	AA==		; Certificate type mnemo
@	CERT	0	0	RSAMD5	AA==		; Algorithm mnemo
@	CERT	0	0	DH	AA==		; Algorithm mnemo
@	CERT	0	0	DSA	AA==		; Algorithm mnemo
@	CERT	0	0	RSASHA1	AA==		; Algorithm mnemo
@	CERT	0	0	DSA-NSEC3-SHA1	AA==		; Algorithm mnemo
@	CERT	0	0	RSASHA1-NSEC3-SHA1	AA==	; Algorithm mnemo
@	CERT	0	0	RSASHA256	AA==		; Algorithm mnemo
@	CERT	0	0	RSASHA512	AA==		; Algorithm mnemo
@	CERT	0	0	ECC-GOST	AA==		; Algorithm mnemo
@	CERT	0	0	ECDSAP256SHA256	AA==		; Algorithm mnemo
@	CERT	0	0	ECDSAP384SHA384	AA==		; Algorithm mnemo
@	CERT	0	0	INDIRECT	AA==		; Algorithm mnemo
@	CERT	0	0	PRIVATEDNS	AA==		; Algorithm mnemo
@	CERT	0	0	PRIVATEOID	AA==		; Algorithm mnemo
@	CERT	0	0	0	Zm8=		; One char padding
@	CERT	0	0	0	Zm9v		; Without padding
@	CERT	0	0	0	Zm9vYg==	; Two base64 blocks
@	CERT	0	0	0	Zm9v YmE=	; Two base64 blocks with blank space between them
@	TYPE37	\# 6 000000000000			; TYPE + Hexadecimal rdata
@	TYPE37	0	0	0	AA==		; TYPE
@	cert	0	0	0	AA==		; Type in lower-case

; KO
@	CERT
@	CERT						; Empty rdata
@	CERT	\# 0					; Hex empty rdata
@	CERT	\#					; Missing hex length
@	CERT	65536	0	0	AA==		; Type overflow
@	CERT	X	0	0	AA==		; Bad type mnemonic
@	CERT	0	65536	0	AA==		; Key tag overflow
@	CERT	0	0	256	AA==		; Algorithm overflow
@	CERT	0	0	0	A		; Continuous block length must be multiple of 4
@	CERT	0	0	0	AB		; Continuous block length must be multiple of 4
@	CERT	0	0	0	ABC		; Continuous block length must be multiple of 4
@	CERT	0	0	0	AA ==		; Continuous block length must be multiple of 4
@	CERT	0	0	0	A===		; Bad padding
@	CERT	0	0	0	=		; Bad padding
@	CERT	0	0	0	==		; Bad padding
@	CERT	0	0	0	===		; Bad padding
@	CERT	0	0	0	====		; Bad padding
@	CERT	0	0	0			; Missing item
@	CERT	\# 6 00000000000000			; Too long rdata
@	CERT	\# 7 000000000000			; Bad rdata length
