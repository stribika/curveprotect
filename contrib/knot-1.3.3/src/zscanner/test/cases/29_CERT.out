OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=FFFFFFFFFF00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000100000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000200000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000300000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000400000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000500000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000600000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000700000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000800000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=00FD00000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=00FE00000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000100
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000200
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000300
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000500
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000600
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000700
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000800
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000A00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000C00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000D00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000E00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=00000000FC00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=00000000FD00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=00000000FE00
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=0000000000666F
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=0000000000666F6F
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=0000000000666F6F62
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=0000000000666F6F6261
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=0025
RDATA=000000000000
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_HEX_CHAR
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_ENUMBER16_OVERFLOW
------
WARNG=ZSCANNER_EBAD_CERT_TYPE
------
WARNG=ZSCANNER_ENUMBER16_OVERFLOW
------
WARNG=ZSCANNER_ENUMBER8_OVERFLOW
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_BASE64_CHAR
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
